spec/ParseServerRESTController.spec.js

```diff
@@ -168,7 +135,8 @@ describe('ParseServerRESTController', () => {
       process.env.MONGODB_STORAGE_ENGINE === 'wiredTiger') ||
     process.env.PARSE_SERVER_TEST_DB === 'postgres'
   ) {
-    describe('transactions', () => {
+    // Randomly fails in CI for Postgres, disabling for now
```

Comment:
disabling means we are not testing transactions at all anymore for postgres, are we sure we want to do that?

Reply:
These tests are essentially the same transaction tests as batch.spec, which I didn’t disable for Postgres. This one just uses the Parse REST controller to test. They both produce the same random connection problem on Postgres, so disabling one increases the probability of the test suite passing without sacrificing integrity

Reply:
Could you maybe add that explanation in short form to the code comment to make it easier for future readers?

Reply:
> Could you maybe add that explanation in short form to the code comment to make it easier for future readers?

Sure, I can add some additional text in a comment, will do so shortly...

An example `batch.spec` passing the transaction tests and the`ParseServerRESTController.spec` failing can be seen in one of the latest commits can be found here: https://github.com/parse-community/parse-server/runs/4692279655?check_suite_focus=true#step:8:2548. Note, this is random, as the batch.spec can fail also, which is the reason I disabled one of them.

Reply:
I am actually not sure we need to disable it. If the tests pass most of the time, we could keep this test and see how it goes?

Reply:
Anecdotally, I've seen the test fail in the suite ~20% of the time (when batch and ParseServerRESTController are enabled). Since there are a number of different Postgres tests in the suite, this can happen more often than not.

It's really how you want to handle it. I disabled because transactions are technically covered in batch.spec for Postgres. If someone adds code that breaks, it will be caught with the batch.spec tests on Mongo/Postgres, or on the  ParseServerRESTController Mongo tests, so I opted for disabling to increase the probability the suite passes. I can enable it back if you want me to?

Reply:
Do you think we have to test transactions in batch, or would it be enough to only test them via REST?

Reply:
That's a good question. I didn't look through the server code in detail, but I suspect everything calls through batch.js, including the REST calls. If this is through, I think your point is to leave the REST tests enabled, because it has to call through batch, which will detect breaks in REST and batch? If so, I can enable REST and disable batch?

Reply:
If we disable one of the two test groups now, and t in the future the remaining transaction tests would be disabled, we should notice that in the coverage report. But since our test structure is largely file-based (each .js file having a corresponding .spec file), we would have to keep both, which in some cases may make sense. I suggest we keep them all enabled for now, and see how it goes.

Reply:
The tests are enabled.